content/blog/2020/07/2020-07-27-machine-learning-plugin-coding-phase2.adoc

```diff
@@ -0,0 +1,160 @@
+---
+layout: post
+title: "Machine Learning Plugin project - Coding Phase 2 blog post"
+tags:
+- machinelearning
+- datascience
+- plugins
+- communitybonding
+- gsoc
+- gsoc2020
+author: loghijiaha
+description: "Coding Phase 2 of Google Summer of Code  2020"
+opengraph:
+  image: /images/gsoc/opengraph.png
+---
+
+image::/images/gsoc/jenkins-gsoc-logo_small.png[align="center"]
+
+Welcome back forks!
+
+This blog post is briefing my coding phase 2 in Jenkins link:/projects/gsoc/2020/projects/machine-learning/[Machine Learning Plugin] for this GSoC 2020.
+
+After successfully passed the evaluation and demo in the phase 1, our team were ahead for facing the challenges in phase 2.
+
+=== Summary
+
+This phase of coding was well spent with fixing lots of bugs and documentation. As the main feature of connecting to a IPython Kernel is done in the phase 1, we were able to focus on fixing minor/major bugs and documenting for the users. According to the JENKINS-62927 issue, a Docker agent was built to facilitate users without concerning plugin dependencies in python. In the act of deprecation of Python 2, we were indeed to port our plugin to support Python 3. We have tested our plugin in conda, venv and windows environments.
```

Comment:
Sentence per line is easier to code review, but not strictly required.  Phrasing changes are all optional.
```suggestion
This phase of coding was well spent by documentation and by fixing many bugs.
As the main feature of connecting to an IPython Kernel is done in phase 1, we were able to focus on fixing minor/major bugs and documenting for the users.
According to the link:https://issues.jenkins-ci.org/browse/JENKINS-62927[JENKINS-62927] issue, a Docker agent was built to facilitate users without concerning plugin dependencies in python.
In the act of deprecation of Python 2, we ported our plugin to support Python 3.
We have tested our plugin in conda, venv and windows environments.
```